include/range/v3/range_concepts.hpp

```diff
@@ -141,6 +142,25 @@ namespace ranges
                     ));
             };
 
+            struct ContiguousRange
+              : refines<RandomAccessRange>
+            {
+                template<typename Rng>
+                using data_reference_t = decltype(*ranges::data(std::declval<Rng&>()));
+
+                template<typename Rng>
+                using datum_t = meta::_t<std::remove_reference<data_reference_t<Rng>>>;
+
+                template<typename Rng>
+                auto requires_(Rng && rng) -> decltype(
+                    concepts::valid_expr(
+                        concepts::model_of<Same, InputRange::value_t<Rng>,
+                            meta::_t<std::remove_cv<datum_t<Rng>>>>(),
+                        concepts::model_of<CommonReference, InputRange::reference_t<Rng>,
+                            data_reference_t<Rng>>()
+                    ));
+            };
```

Comment:
Yes, that's the intended evolution. `data` already accepts the class of iterators that we know are contiguous - pointers! - we would literally replace `is_pointer<foo>` with `ContiguousIterator<foo>`.